Parallelize storage of incremental segments (#13982)
apache/druid: docs/development/extensions-core/kafka-supervisor-reference.md

| `maxPendingPersists`              | Integer        | Maximum number of persists that can be pending but not started. If this limit would be exceeded by a new intermediate persist, ingestion will block until the currently-running persist finishes. Maximum heap memory usage for indexing scales with `maxRowsInMemory` * (2 + `maxPendingPersists`).                                                                                                                                                                                                                                                                                                                                                    | no (default == 0, meaning one persist can be running concurrently with ingestion, and none can be queued up) |

> |  numPersistThreads                | Integer        | The number of threads to use to create and persist incremental segments on the disk. Higher ingestion data throughput results in larger number of incremental segments, causing significant cpu time to be spent on the creation of the incremental segments on the disk. For datasources with number of columns running into hundreds or thousands, creation of the incremental segments may take up significant time, in the order of multiple seconds. Both these scenarios can cause ingestion can pause frequently or stall causing it to fall behind. With more threads the segment creation can be parallelized without blocking ingestion as long as there are sufficient cpu resources available.                                                         | no (default == 1)                                          |

Reviewer: ```suggestion
|  `numPersistThreads`                | Integer        | The number of threads to use to create and persist incremental segments on the disk. Higher ingestion data throughput results in a larger number of incremental segments, causing significant cpu time to be spent on the creation of the incremental segments on the disk. For datasources with number of columns running into hundreds or thousands, creation of incremental segments may take up significant time, in the order of multiple seconds. In both of these scenarios, ingestion can stall or pause frequently, causing it to fall behind. You can use additional threads to parallelize the segment creation without blocking ingestion as long as there are sufficient cpu resources available.                                                         | no (default == 1)                                          |
```

Author: Incorporated feedback, thanks @ektravel
